forge/src/executor/db/cache.rs

```diff
@@ -0,0 +1,144 @@
+use bytes::Bytes;
```

Comment:
This file is mostly copied from upstream `revm` except I also implement `DatabaseRef` on the DB for `inspect_ref` and `transact_ref`. I thought it would be a good idea to have our own DB implementation as it will probably change anyway when we port forking

Reply:
Do we need to copy over the cache etc code? Can we not wrap around an existing backend to implement forking?

Reply:
Thinking about it more, probably not. I'll get the `DatabaseRef` impl upstreamed

Reply:
Opened up a PR upstream, will remove this and use the upstream version in a follow up PR 

https://github.com/bluealloy/revm/pull/56